app/helpers/application_helper.rb

```diff
@@ -34,4 +34,16 @@ def decorated_session
       @_decorated_session ||= SessionDecorator.new
     end
   end
+
+  def sp_session
+    session[:sp]
+  end
+
+  def loa1_signup_context?
+    sp_session && current_user && !current_user.recovery_code.present?
```

Comment:
This means that the last chance a user has to cancel is on the phone confirmation page. Once they confirm their phone number, the Cancel link is no longer available in the Recovery Code page. Is that intentional?